test/meta.cpp

```diff
@@ -190,4 +194,263 @@ BOOST_AUTO_TEST_CASE(ForwardLike) {
   // clang-format on
 }
 
+BOOST_AUTO_TEST_CASE(FunctionTraits)
+{
+  {
+    using F = int(float);
```

Comment:
Could you please define *real* functions, and do `decltype`? Predefining function type as `using F = ...` is much like a short-cut, which could be introducing another layer of inaccuracy on the context of unit testing.